### ISSUE TITLE
[CI] CancelTests testUpdateByQueryCancelWithWorkers fails

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#28660]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.2+openjdk-java9-periodic/68/console

```
./gradlew :modules:reindex:test \
  -Dtests.seed=59581F77658F2744 \
  -Dtests.class=org.elasticsearch.index.reindex.CancelTests \
  -Dtests.method="testUpdateByQueryCancelWithWorkers" \
  -Dtests.security.manager=true \
  -Dtests.locale=dua-CM \
  -Dtests.timezone=Asia/Samarkand
```

```
03:53:36    > Throwable #1: java.lang.AssertionError: Didn't find at least one sub task that was cancelled
03:53:36    > 	at __randomizedtesting.SeedInfo.seed([59581F77658F2744:96A721FC29CCF116]:0)
03:53:36    > 	at org.elasticsearch.index.reindex.CancelTests.testCancel(CancelTests.java:146)
03:53:36    > 	at org.elasticsearch.index.reindex.CancelTests.testUpdateByQueryCancelWithWorkers(CancelTests.java:253)
03:53:36    > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
03:53:36    > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
03:53:36    > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
03:53:36    > 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
```

Unfortunately I wasn't able to reproduce. I found https://github.com/elastic/elasticsearch/issues/26758 but that one seems to be a timeout while this error message looks slightly different. Will start looking into this.

**Comment by nik9000** [comment]
I believe this is the same issue, yeah. Though, I believe I didn't backport the changes that I made for #26758 to the 6.2 branch so that is probably what is up here. 6.2 was about to go out the door and I didn't want to destabilize it.

**Comment by cbuescher** [comment]
I took a look at the above mentioned fix for #26758 and it looks like this indeed might also solve this. I ported the changes of @nik9000 from 6.x back to 6.2 (aadca4535f16bdc9c6d39c09ddce9cbc382c2201) and will close for now, we can reopen if this turns out that this is not enough.
